<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>sct.seq</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="sct-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="sct-module.html">Package&nbsp;sct</a> ::
        Module&nbsp;seq
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="sct.seq-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== MODULE DESCRIPTION ==================== -->
<h1 class="epydoc">Module seq</h1><p class="nomargin-top"><span class="codelink"><a href="sct.seq-pysrc.html">source&nbsp;code</a></span></p>
<p>Module that deals with sequence properties within SCT. Loads in yaml 
  files containing scattering parameters and residue volumes</p>

<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">dictionary</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#residue_freq_dict" class="summary-sig-name">residue_freq_dict</a>()</span><br />
      Create dictionary holding the frequencies of all residues</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#residue_freq_dict">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">list, dictionary</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#parse_fasta" class="summary-sig-name">parse_fasta</a>(<span class="summary-sig-arg">filename</span>)</span><br />
      Parse fasta files.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#parse_fasta">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">dictionary</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#fasta_res_freq" class="summary-sig-name">fasta_res_freq</a>(<span class="summary-sig-arg">filename</span>)</span><br />
      Read fasta file and return dictionary of residue frequencies.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#fasta_res_freq">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">dictionary</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#seq_file_to_freq" class="summary-sig-name">seq_file_to_freq</a>(<span class="summary-sig-arg">seq_filename</span>,
        <span class="summary-sig-arg">file_type</span>)</span><br />
      Read a sequence file in either fasta or YAML format and return 
      residue frequencies.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#seq_file_to_freq">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">float</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#calc_hydration_volume" class="summary-sig-name">calc_hydration_volume</a>(<span class="summary-sig-arg">res_freq</span>)</span><br />
      Estimate hydrated volume of a protein from its amino acid composition</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#calc_hydration_volume">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">float, float</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#calc_hydration_effect" class="summary-sig-name">calc_hydration_effect</a>(<span class="summary-sig-arg">res_freq</span>)</span><br />
      Calculates the effect of hydration on protein volume</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#calc_hydration_effect">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">integer</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#sum_res_no" class="summary-sig-name">sum_res_no</a>(<span class="summary-sig-arg">resids</span>,
        <span class="summary-sig-arg">res_freq</span>)</span><br />
      Get the total number of residues for selected residue codes</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#sum_res_no">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">float</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#sum_mass" class="summary-sig-name">sum_mass</a>(<span class="summary-sig-arg">resids</span>,
        <span class="summary-sig-arg">res_freq</span>)</span><br />
      Calculate the total mass of the input residue constitution</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#sum_mass">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">float</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#sum_volume" class="summary-sig-name">sum_volume</a>(<span class="summary-sig-arg">resids</span>,
        <span class="summary-sig-arg">res_freq</span>,
        <span class="summary-sig-arg">dataset</span>)</span><br />
      Calculate the total volume of the input residue constitution</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#sum_volume">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">float</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#spec_volume" class="summary-sig-name">spec_volume</a>(<span class="summary-sig-arg">resids</span>,
        <span class="summary-sig-arg">res_freq</span>,
        <span class="summary-sig-arg">dataset</span>)</span><br />
      Calculate the specific volume of the input residue constitution</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#spec_volume">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">float</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#calc_absorption_coeffs" class="summary-sig-name">calc_absorption_coeffs</a>(<span class="summary-sig-arg">res_freq</span>,
        <span class="summary-sig-arg">mass</span>)</span><br />
      Calculate absorption coefficients</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#calc_absorption_coeffs">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">float</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#sum_b" class="summary-sig-name">sum_b</a>(<span class="summary-sig-arg">resids</span>,
        <span class="summary-sig-arg">res_freq</span>,
        <span class="summary-sig-arg">heavy_water</span>)</span><br />
      Calculate total scattering length of selected residues</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#sum_b">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">integer</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="sct.seq-module.html#sum_electrons" class="summary-sig-name">sum_electrons</a>(<span class="summary-sig-arg">resids</span>,
        <span class="summary-sig-arg">res_freq</span>)</span><br />
      Calculate the total number of elections in the selected residues</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="sct.seq-pysrc.html#sum_electrons">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="share_dir"></a><span class="summary-name">share_dir</span> = <code title="os.path.join(__file__.rsplit(os.sep, 1) [0], 'share')">os.path.join(__file__.rsplit(os.sep, 1) [0], 'share')</code><br />
      Path to the input data files
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="sct.seq-module.html#polar" class="summary-name">polar</a> = <code title="['ARG', 'ASN', 'ASP', 'GLN', 'GLU', 'HIS', 'LYS', 'SER', 'THR']">['ARG', 'ASN', 'ASP', 'GLN', 'GLU', 'HIS', 'LYS', 'SER<code class="variable-ellipsis">...</code></code><br />
      List of polar amino acids (aa) - 3 letter codes
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="sct.seq-module.html#non_polar" class="summary-name">non_polar</a> = <code title="['ALA', 'CYS', 'GLY', 'ILE', 'LEU', 'MET', 'PHE', 'PRO', 'TRP', 'TYR',\
 'VAL']">['ALA', 'CYS', 'GLY', 'ILE', 'LEU', 'MET', 'PHE', <code class="variable-ellipsis">...</code></code><br />
      List of non-polar aa - 3 letter codes
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="amino_acids"></a><span class="summary-name">amino_acids</span> = <code title="polar+ non_polar">polar+ non_polar</code><br />
      List of all aa - 3 letter codes
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="sct.seq-module.html#monosaccharides" class="summary-name">monosaccharides</a> = <code title="['FUC', 'GAL', 'GLC', 'MAN', 'NAG', 'NGA', 'SIA']">['FUC', 'GAL', 'GLC', 'MAN', 'NAG', 'NGA', '<code class="variable-ellipsis">...</code></code><br />
      list of monosaccharides - 3 letter codes
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="all_residues"></a><span class="summary-name">all_residues</span> = <code title="amino_acids+ monosaccharides">amino_acids+ monosaccharides</code><br />
      List combining amino_acids and monosaccharides - 3 letter codes
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="aa1"></a><span class="summary-name">aa1</span> = <code title="list(&quot;ARNDCQEGHILKMFPSTWYV&quot;)">list(&quot;ARNDCQEGHILKMFPSTWYV&quot;)</code><br />
      List of amino acid 1 letter codes
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="aa1to3"></a><span class="summary-name">aa1to3</span> = <code title="dict(zip(aa1, sorted(amino_acids)))">dict(zip(aa1, sorted(amino_acids)))</code><br />
      Dictionary providing 1 to 3 letter aa code conversion
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="sct.seq-module.html#param_file" class="summary-name">param_file</a> = <code title="file(os.path.join(share_dir, 'sluv_parameters.yml'), 'r')">file(os.path.join(share_dir, 'sluv_parameters.yml<code class="variable-ellipsis">...</code></code><br />
      Path to file containing scattering and mass parameters
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="params"></a><span class="summary-name">params</span> = <code title="yaml.load(param_file)">yaml.load(param_file)</code><br />
      Dictionary of scattering parameters read in from file
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="vol_file"></a><span class="summary-name">vol_file</span> = <code title="file(os.path.join(share_dir, 'aa_volumes.yml'), 'r')">file(os.path.join(share_dir, 'aa_volumes.yml'), 'r')</code><br />
      Path to file containing volumes parameters
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="res_vols"></a><span class="summary-name">res_vols</span> = <code title="yaml.load(vol_file)">yaml.load(vol_file)</code><br />
      Dictionary of residue volumes read in from file
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="bDH_diff"></a><span class="summary-name">bDH_diff</span> = <code title="params ['solvent'] ['BOD']-params ['solvent'] ['BOH']">params ['solvent'] ['BOD']-params ['solvent'] ['BOH']</code><br />
      Difference in scattering length for heavy and light water
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="residue_freq_dict"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">residue_freq_dict</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#residue_freq_dict">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Create dictionary holding the frequencies of all residues</p>
  <dl class="fields">
    <dt>Returns: dictionary</dt>
        <dd>Initialized dictionary for all residue types recognized by SCT 
          (three letter residue codes are used as the key). Values for each
          residue type are set to 0.</dd>
  </dl>
</td></tr></table>
</div>
<a name="parse_fasta"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">parse_fasta</span>(<span class="sig-arg">filename</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#parse_fasta">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Parse fasta files. Adapted from: 
  http://www.petercollingridge.co.uk/python-bioinformatics-tools/fasta-parser</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>filename</code></strong> (string) - Path to fasta file to be read.</li>
    </ul></dd>
    <dt>Returns: list, dictionary</dt>
        <dd>1. Sequence identifiers found in order they appear in the file
          <ol start="2">
            <li>
              Dictionary where keys are sequence identifiers and values are
              sequences as strings (single letter amino acid codes used).
            </li>
          </ol></dd>
  </dl>
</td></tr></table>
</div>
<a name="fasta_res_freq"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">fasta_res_freq</span>(<span class="sig-arg">filename</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#fasta_res_freq">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Read fasta file and return dictionary of residue frequencies.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>filename</code></strong> (string) - Path to fasta file to be read.</li>
    </ul></dd>
    <dt>Returns: dictionary</dt>
        <dd>Dictionary for all residue types recognized by SCT (three letter 
          residue codes are used as the key). Values are the frequency with
          which each residue is found in input fasta file.</dd>
  </dl>
</td></tr></table>
</div>
<a name="seq_file_to_freq"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">seq_file_to_freq</span>(<span class="sig-arg">seq_filename</span>,
        <span class="sig-arg">file_type</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#seq_file_to_freq">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Read a sequence file in either fasta or YAML format and return residue
  frequencies.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>seq_filename</code></strong> (string) - Path to sequence file (either fasta or YAML)</li>
        <li><strong class="pname"><code>file_type</code></strong> (string) - Is the input a fasta ('fas') or YAML ('yml') file</li>
    </ul></dd>
    <dt>Returns: dictionary</dt>
        <dd>Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</dd>
  </dl>
</td></tr></table>
</div>
<a name="calc_hydration_volume"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">calc_hydration_volume</span>(<span class="sig-arg">res_freq</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#calc_hydration_volume">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Estimate hydrated volume of a protein from its amino acid 
  composition</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
    </ul></dd>
    <dt>Returns: float</dt>
        <dd>Estimated total volume of a protein containing the residue 
          frequencies input + a hydration shell of water.</dd>
  </dl>
</td></tr></table>
</div>
<a name="calc_hydration_effect"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">calc_hydration_effect</span>(<span class="sig-arg">res_freq</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#calc_hydration_effect">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the effect of hydration on protein volume</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
    </ul></dd>
    <dt>Returns: float, float</dt>
        <dd>Difference in total hydrated and un-hydrated volumes.
          <p>Equivalent number of bound H2O to the volume difference.</p></dd>
  </dl>
</td></tr></table>
</div>
<a name="sum_res_no"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">sum_res_no</span>(<span class="sig-arg">resids</span>,
        <span class="sig-arg">res_freq</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#sum_res_no">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Get the total number of residues for selected residue codes</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>resids</code></strong> (list) - List of three letter residue codes.</li>
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
    </ul></dd>
    <dt>Returns: integer</dt>
        <dd>Total number of residues of the types given in resids.</dd>
  </dl>
</td></tr></table>
</div>
<a name="sum_mass"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">sum_mass</span>(<span class="sig-arg">resids</span>,
        <span class="sig-arg">res_freq</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#sum_mass">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate the total mass of the input residue constitution</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>resids</code></strong> (list) - List of three letter residue codes</li>
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
    </ul></dd>
    <dt>Returns: float</dt>
        <dd>Total mass of residues of the types given in resids (i.e. 
          sum[mass of residue type * frequency] over all input residue 
          types).</dd>
  </dl>
</td></tr></table>
</div>
<a name="sum_volume"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">sum_volume</span>(<span class="sig-arg">resids</span>,
        <span class="sig-arg">res_freq</span>,
        <span class="sig-arg">dataset</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#sum_volume">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate the total volume of the input residue constitution</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>resids</code></strong> (list) - List of three letter residue codes</li>
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
        <li><strong class="pname"><code>dataset</code></strong> (string) - Choice from the read in residue volume datasets - default list: 
          cohn1943, zamyatin1972, zamyatin1984, chothia1975, perkins1986a, 
          richards1974 &amp; perkins1986b.</li>
    </ul></dd>
    <dt>Returns: float</dt>
        <dd>Total volume of residues of the types given in resids - 
          sum(volume of residue type * frequency) over all input residue 
          types.</dd>
  </dl>
</td></tr></table>
</div>
<a name="spec_volume"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">spec_volume</span>(<span class="sig-arg">resids</span>,
        <span class="sig-arg">res_freq</span>,
        <span class="sig-arg">dataset</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#spec_volume">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate the specific volume of the input residue constitution</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>resids</code></strong> (list) - List of three letter residue codes</li>
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
        <li><strong class="pname"><code>dataset</code></strong> (string) - Choice from the read in residue volume datasets - default list: 
          cohn1943, zamyatin1972, zamyatin1984, chothia1975, perkins1986a, 
          richards1974 &amp; perkins1986b.</li>
    </ul></dd>
    <dt>Returns: float</dt>
        <dd>Specific volume of the input residue constitution - sum(mass of 
          residue type * frequency) * avagadros number / sum(mass of 
          residue type * frequency). Where sums are over all input residue 
          types.</dd>
  </dl>
</td></tr></table>
</div>
<a name="calc_absorption_coeffs"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">calc_absorption_coeffs</span>(<span class="sig-arg">res_freq</span>,
        <span class="sig-arg">mass</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#calc_absorption_coeffs">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate absorption coefficients</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
        <li><strong class="pname"><code>mass</code></strong> (float) - Total mass of the protein</li>
    </ul></dd>
    <dt>Returns: float</dt>
        <dd>Absorption coefficient</dd>
  </dl>
</td></tr></table>
</div>
<a name="sum_b"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">sum_b</span>(<span class="sig-arg">resids</span>,
        <span class="sig-arg">res_freq</span>,
        <span class="sig-arg">heavy_water</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#sum_b">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate total scattering length of selected residues</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>resids</code></strong> (list) - List of three letter residue codes</li>
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
        <li><strong class="pname"><code>heavy_water</code></strong> (boolean) - Are we calculating for deuterated system?</li>
    </ul></dd>
    <dt>Returns: float</dt>
        <dd>Total scattering length for the selected residues.</dd>
  </dl>
</td></tr></table>
</div>
<a name="sum_electrons"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">sum_electrons</span>(<span class="sig-arg">resids</span>,
        <span class="sig-arg">res_freq</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="sct.seq-pysrc.html#sum_electrons">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate the total number of elections in the selected residues</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>resids</code></strong> (list) - List of three letter residue codes</li>
        <li><strong class="pname"><code>res_freq</code></strong> (dictionary) - Dictionary for residue type frequency. Three letter residue code 
          for the key and residue type frequency as the values.</li>
    </ul></dd>
    <dt>Returns: integer</dt>
        <dd>Total number of electrons in the selected residues</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== VARIABLES DETAILS ==================== -->
<a name="section-VariablesDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-VariablesDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="polar"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">polar</h3>
  <p>List of polar amino acids (aa) - 3 letter codes</p>
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
['ARG', 'ASN', 'ASP', 'GLN', 'GLU', 'HIS', 'LYS', 'SER', 'THR']
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<a name="non_polar"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">non_polar</h3>
  <p>List of non-polar aa - 3 letter codes</p>
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
['ALA', 'CYS', 'GLY', 'ILE', 'LEU', 'MET', 'PHE', 'PRO', 'TRP', 'TYR',<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
 'VAL']
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<a name="monosaccharides"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">monosaccharides</h3>
  <p>list of monosaccharides - 3 letter codes</p>
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
['FUC', 'GAL', 'GLC', 'MAN', 'NAG', 'NGA', 'SIA']
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<a name="param_file"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">param_file</h3>
  <p>Path to file containing scattering and mass parameters</p>
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
file(os.path.join(share_dir, 'sluv_parameters.yml'), 'r')
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="sct-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Mon Jun 16 11:53:31 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
